sql/2019/04_Media/04_09a.sql

```diff
@@ -1,22 +1,41 @@
-#StandardSQL
-/*
-standard SQL
-04_09a
-112 GB
+#standardSQL
+# 04_09a: Client Hints
+CREATE TEMP FUNCTION hasClientHintMeta(payload STRING)
+RETURNS BOOLEAN LANGUAGE js AS '''
+try {
+  var $ = JSON.parse(payload);
+  var almanac = JSON.parse($._almanac);
+  return !!almanac['meta-nodes'].find(meta => meta['http-equiv'].toLowerCase() == 'accept-ch');
+} catch (e) {
+  return null;
+}
+''';
 
-this gets the requests whose servers add a "accept-CH" header
-I'll do another query for the meta tag
-
-*/
-
-select count(host)
-from(
-SELECT url, 
-	NET.HOST(url)AS  host, 
-	respOtherHeaders
-
-
-FROM `summary_requests.2019_07_01_*`
-WHERE respOtherHeaders LIKE "%Accept-CH%"
-ORDER BY HOST ASC
-)
\ No newline at end of file
+SELECT
+  client,
+  accept_ch_header,
+  accept_ch_meta,
+  total,
+  ROUND(accept_ch_header * 100 / total, 2) AS pct_accept_ch_header,
+  ROUND(accept_ch_meta * 100 / total, 2) AS pct_accept_ch_meta
```

Comment:
Add pct for pages accepting CH at all (has a ch header or meta tag)

Reply:
Done